### PR TITLE
introduce a DtypeConfig

### PR DESCRIPTION
**Pull request by iwknow**
introduce a DtypeConfig system for granular control over JAX dtypes in model components, configurable via server arguments.

## Motivation
see: https://github.com/sgl-project/sglang-jax/issues/661

## Modifications

introduce a DtypeConfig system for granular control over JAX dtypes in model components, configurable via server arguments.

This is the first PR of a series of PRs that will introduce DtypeConfig to sglang-jax. see design doc at https://github.com/sgl-project/sglang-jax/issues/798 

## Accuracy Tests

N/A

## Benchmarking and Profiling

N/A

## Checklist

- [x] Please use English, otherwise it will be closed.
- [x] The purpose of the PR, or link existing issues this PR will resolve.
- [x] The test plan, such as providing test command.
- [N/A] (Optional) The necessary documentation update.

### CODE REVIEW

**Comment by iwknow** [comment]
> @iwknow For this small feature branch, I think it is fine. Wait for your opinion about epic development.

if you want to reduce review burden, i can squash small changes into one PR. That reduces the number of PR to review but each PR is larger and harder to review. smaller PR contains more focused code change and easier to review. i can do both. what's your preference? any way a separate branch is unnecessary in my opinion.

if you don't want the feature to be partially released, i think developing in a separate branch is a good idea.

**Comment by iwknow** [comment]
@aolemila  can you please create a new branch `feat/dtype_config` for me or provide me the related permissions so that i can push all my changes there. thank you so much!

**Comment by iwknow** [comment]
@aolemila please review this PR along with https://github.com/sgl-project/sglang-jax/pull/855. This is the foundation for other changes. It sets the direction of the following PRs. Therefore, i want to get your opinion before I went too far. once this looks good to you, i will check in the request of the code and the next time you need to review is when i modify one of the model and execute the verification steps described in  the design doc. thanks!

**Comment by aolemila** [comment]
> @aolemila please review this PR along with #855. This is the foundation for other changes. It sets the direction of the following PRs. Therefore, i want to get your opinion before I went too far. once this looks good to you, i will check in the request of the code and the next time you need to review is when i modify one of the model and execute the verification steps described in the design doc. thanks!

I have merged both of them.
